app/pages/household/[houseId]/utility/[utilityId]/index.tsx

```diff
@@ -27,47 +27,163 @@ const Login = () => {
+        setValidity(false);
+        console.log("BOOKING INVALID");
       }
-    );
+    }
+  }, [bookingData]); // Unsure what to do with this
+
+  const handleBooking = async (event: any) => {
+    // Prevents page from auto reloading every form submit
+    event.preventDefault();
+
+    const bookingDesc = event.target.elements.desc.value;
+    const bookingTo = event.target.elements.to.value;
+    const bookingFrom = event.target.elements.from.value;
+    setBookingData({ bookingDesc, bookingTo, bookingFrom });
+
+    if (amenityData) {
+      // Checks if amenity is already booked
+      var utilityAvail = isUtilityAvail(
+        bookingFrom,
+        bookingTo
+        // amenityData
+      );
+      // Checks if user is booking into the past
+      var bookingMakesSense = datesWellFormed(bookingFrom, bookingTo);
+
+      console.log("UTIL AVAIL?", utilityAvail);
+      console.log("BOOKING GOOD?", bookingMakesSense);
 
-    console.log("Document written with ID: ", docRef.id);
+      // Do below if ((bookingMakesSense && alreadyBooked) == true)
+      if (bookingMakesSense == true && utilityAvail == true) {
+        console.log("HELLO WORLDDDDDDD");
+        const docRef = await addDoc(bookingRef, {
+          utilityId,
+          desc: bookingDesc,
+          userId: currentUser?.userId,
+          householdId: houseId,
+          from: bookingFrom,
+          to: bookingTo,
+        });
+
+        const updateAmenity = await updateDoc(amenityRef, {
+          "latestBooking.from": bookingFrom,
+          "latestBooking.to": bookingTo,
+          "latestBooking.userId": currentUser?.userId,
+          "latestBooking.bookingID": docRef.id,
+        });
+        setValidity(true);
+
+        // ToDo:
+        // 1. Change last booking system to an array of bookings?
+        //    Will have to further check validity in a loop maybe
+        // 2. Implement delete booking feature
+      }
+    }
+    // console.log("Document wbookingFromritten with ID: ", docRef.id);
   };
 
-  const isAmenityBooked = (data: DocumentData): boolean => {
-    const now = new Date();
-    const booking = data?.latestBooking?.to.toDate();
-    console.log("day now", data?.name, now.getDay(), booking?.getDay());
-    if (now.getDay() == booking?.getDay()) {
+  // Checks if utility is currently booked
+  // Returns true if utility is available
+  // Current booking date is equal or less to the latest booking toDate
+  const isUtilityAvail = (
+    bookingFrom: any,
+    bookingTo: any
+    // amenityData: DocumentData
+  ): boolean => {
+    // if amenityData is undefined (no amenity?) then return true as no prev booking
+    if (
+      amenityData == undefined ||
+      bookingFrom == undefined ||
+      bookingTo == undefined
+    ) {
       return true;
     }
 
+    // get latestBookingFrom and latestBookingTo from UtilityID
+    const LBFrom = new Date(amenityData?.latestBooking.from);
+    const LBTo = new Date(amenityData?.latestBooking.to);
```

Comment:
This seems to be causing an error (undefined variable).
I think you need to create a latestBooking (to and from) field for the amenity when you create the amenity because right now I think theres only: name, desc and type!